radixdlt/src/integration/java/com/radixdlt/consensus/deterministic/ControlledNetwork.java

```diff
@@ -49,19 +51,54 @@
  * This class is not thread safe.
  */
 public final class ControlledNetwork {
-	private final ImmutableList<ECPublicKey> nodes;
-	private final ImmutableMap<ChannelId, LinkedList<ControlledMessage>> messageQueue;
-
-	ControlledNetwork(ImmutableList<ECPublicKey> nodes) {
-		this.nodes = nodes;
-		this.messageQueue = nodes.stream()
-			.flatMap(n0 -> nodes.stream().map(n1 -> new ChannelId(n0, n1)))
-			.collect(
-				ImmutableMap.toImmutableMap(
-					key -> key,
-					key -> new LinkedList<>()
-				)
-			);
+	// Process sync related messages before consensus messages
+	private static final MessageRank EARLIEST_POSSIBLE = new MessageRank(0L, 0L);
+
+	private final MessageQueue messageQueue = new MessageQueue();
+
+	ControlledNetwork() {
+		// Nothing here right now
+	}
+
+	// Message ranking.  Used to implement timeouts.
+	// Messages in a particular ranking are processed in arrival order, but
+	// timeouts in particular will be put into the next ranking to ensure
+	// that they are processed after the current ranking. The rank is changed
+	// for each node whenever a new view or epoch change is seen.
+	static final class MessageRank implements Comparable<MessageRank> {
```

Comment:
Is `MessageRank` some known term? The naming does not do it for me :), tells me nothing reading the name. 

I guess we want it to be an implementation detail that it is using Epochs and View number? And semantically it represents some ordering?

So why not call it `MessageOrdering` or `AbstractTime`? Since in fact it does not hold any "message" it is just used to order some abstraction of "time"? Or `AbstractOrder`? :)

Reply:
Not sure about the confusion here. It simply is a class which represents the rank of a message. `MessageOrdering` would imply more of a `Comparator`. Anyway, this class isn't actually a part of this PR.

Reply:
> MessageOrdering would imply more of a Comparator
Well yes, that is basically the only thing that the class does right now, with code:

```java
private static final Comparator<MessageRank> COMPARATOR =
			Comparator.comparingLong((MessageRank eav) -> eav.epoch).thenComparingLong(eav -> eav.view);
```

Right? :)

I'm saying that "Rank" (which really should be "Ranking" ) and that "Message" is confusing for me. It does not contain any message, and it can be used for anything really, not only messages. And ranking implies some more sophisticated ranking logic, e.g. ELO Ranking, but we only really compare Epoch+View and is only really used for **_ordering_** ??

Even if not part of PR, I think the naming still can be improved  IMO :)

Reply:
It represents the rank of a message.  Not sure how better to communicate this?